### PR TITLE
Auto Tool for Nuker Module

### PR DESCRIPTION
**Pull request by MrDe0**
Adding autotool feature for nuker module.

### CODE REVIEW

File: Horion/Module/Modules/Nuker.h [hunk 1]
```diff
@@ -7,6 +7,8 @@ class Nuker : public IModule {
 	int nukerRadius = 4;
 	bool veinMiner = false;
 	bool autodestroy = true;
+	bool autotool = true;
+	void findTool();
```
Comment (N0tExisting) [inline Horion/Module/Modules/Nuker.h:11]: Maybe find other tools as well (shovel/axe)

File: Horion/Module/Modules/Nuker.cpp [hunk 2]
```diff
@@ -23,9 +24,29 @@ void Nuker::onTick(C_GameMode* gm) {
...
+	for (int n = 0; n < 9; n++) {
+		C_ItemStack* stack = inv->getItemStack(n);
+
+		if (stack->item != nullptr) {
+			if ((*stack->item)->isMiningTool()) {
+				slot = n;
+			}
+		}
+	}
+	supplies->selectedHotbarSlot = slot;
```
Comment (N0tExisting) [inline Horion/Module/Modules/Nuker.cpp:51]: Make an `int prevSlot` and select that later at the end of `Nuker::onTick`

File: Horion/Module/Modules/Nuker.cpp [hunk 3]
```diff
@@ -23,9 +24,29 @@ void Nuker::onTick(C_GameMode* gm) {
...
+	C_PlayerInventoryProxy* supplies = g_Data.getLocalPlayer()->getSupplies();
+	C_Inventory* inv = supplies->inventory;
+
+	int slot = supplies->selectedHotbarSlot;
+	for (int n = 0; n < 9; n++) {
+		C_ItemStack* stack = inv->getItemStack(n);
+
+		if (stack->item != nullptr) {
+			if ((*stack->item)->isMiningTool()) {
+				slot = n;
```
Comment (N0tExisting) [inline Horion/Module/Modules/Nuker.cpp:47]: I think we might need a [`MobEquipmentPacket`](https://github.com/horionclient/Horion/blob/master/SDK/CPacket.h#L74) if we are changing more than once per tick

**Review by N0tExisting — COMMENTED** [review]
There is also something in [Hooks.cpp](https://github.com/horionclient/Horion/blob/master/Memory/Hooks.cpp#L1205), which would allow you to make a module (the only problem would be that you would have to hardcode the preferred tool for each block or find the function for that(Maybe Something w/ [C_Material](https://github.com/horionclient/Horion/blob/master/SDK/CBlockLegacy.h#L9))).

File: Horion/Module/Modules/Nuker.h (hunk 1 above)
Comment (d7do7e) [reply Horion/Module/Modules/Nuker.h:11]: yah i did that already then i face a problem which is i dont know if he is breaking a block that use a picaxe or shovel so i just stick with miningtool.

bool isMiningTool(void) {
		if (itemId == 318) return true;                                                                                       // Diamond Picaxe
		if (itemId == 297) return true;                                                                                       // Iron Picaxe
		if (itemId == 324) return true;                                                                                       // Gold Picaxe
		if (itemId == 314) return true;                                                                                       // Stone Picaxe
		if (itemId == 310) return true;                                                                                       // Wood Picaxe
			return false;
	}
	//bool isShovelingTool(void){
	//	if (itemId == ) return true;                                                                                       // Diamond Shovel 
	//	if (itemId == 256) return true;                                                                                       // Iron Shovel 
	//	if (itemId == 273) return true;                                                                                       // Stone Shovel 
	//	if (itemId == 269) return true;                                                                                       // Wood Shovel 
	//	return false;	
	//}
	//bool isWoodworkingTool(void){
	//	if (itemId == 279) return true;                                                                                      // Diamond Axe 
	//	if (itemId == 258) return true;                                                                                      // Iron Axe 
	//	if (itemId == 275) return true;                                                                                      // Stone Axe 
	//	if (itemId == 271) return true;                                                                                      // Wood Axe  
	//	return false;
	//}

File: Horion/Module/Modules/Nuker.cpp (hunk 3 above)
Comment (d7do7e) [reply Horion/Module/Modules/Nuker.cpp:47]: i dont get it how we could change more than once per tick ? 🙄

File: Horion/Module/Modules/Nuker.cpp (hunk 2 above)
Comment (d7do7e) [reply Horion/Module/Modules/Nuker.cpp:51]: what the difference that could make?

u mean after nuker is turned off u want the prevSlot to be reselected? 
if so then i dont understand the point of that 😕

**Comment by MrDe0** [comment]
> There is also something in [Hooks.cpp](https://github.com/horionclient/Horion/blob/master/Memory/Hooks.cpp#L1205), which would allow you to make a module (the only problem would be that you would have to hardcode the preferred tool for each block or find the function for that(Maybe Something w/ [C_Material](https://github.com/horionclient/Horion/blob/master/SDK/CBlockLegacy.h#L9))).

woo that's interesting !

**Comment by MrDe0** [comment]
all right then i am gonna explore that hooks.cpp and see what i could do with it .

**Comment by N0tExisting** [comment]
Also i forgot (to mention) that `onTick` is called [after the tick](https://github.com/horionclient/Horion/blob/master/Memory/Hooks.cpp#L326) (if you make ur own method, then call it before it is possible to implement what i was saying)

**Pull request closed without merging**